tests/acl/test_acl_outer_vlan.py

```diff
@@ -203,77 +240,53 @@ def setup_vlan(rand_selected_dut, vlan_setup_info):
     time.sleep(10)
 
 
-def teardown_vlan(rand_selected_dut, vlan_setup_info):
-    """
-    Remove testing vlan
-    """
-    logger.info("Removing Vlan for testing")
-    vlan_setup, test_ports = vlan_setup_info
-    # Remove ports from test vlan
-    cmds = []
-    for new_vlan in TEST_VLAN_LIST:
-        tagged_port = vlan_setup[new_vlan].get('tagged_ports', None)
-        untagged_port = vlan_setup[new_vlan].get('untagged_ports', None)
-        if tagged_port:
-            cmds.append("config vlan member del {} {}".format(new_vlan, tagged_port[0]))
-        if untagged_port:
-            cmds.append("config vlan member del {} {}".format(new_vlan, untagged_port[0]))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-    # Remove IP from Vlan interface
-    cmds = []
-    for new_vlan in TEST_VLAN_LIST:
-        cmds.append('config interface ip remove Vlan{} {}'.format(new_vlan, vlan_setup[new_vlan]['vlan_ip'][IPV4]))
-        cmds.append('config interface ip remove Vlan{} {}'.format(new_vlan, vlan_setup[new_vlan]['vlan_ip'][IPV6]))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-    # Remove testing vlan
-    cmds = []
-    for new_vlan in TEST_VLAN_LIST:
-        cmds.append('config vlan del {}'.format(new_vlan))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-    default_vlan_id = vlan_setup['default_vlan'].replace("Vlan", "")
-    # Add back interface to default Vlan
-    cmds = []
-    for port in list(test_ports.keys()):
-        cmds.append('config vlan member add {} {} --untagged'.format(default_vlan_id, port))
-    rand_selected_dut.shell_cmds(cmds=cmds)
-    time.sleep(10)
-
-
 @pytest.fixture(scope='module', autouse=True)
-def vlan_setup_teardown(rand_selected_dut, ptfadapter, vlan_setup_info, tbinfo):
+def vlan_setup_teardown(rand_selected_dut, vlan_setup_info, duthost, ptfhost):
     try:
-        setup_vlan(rand_selected_dut, vlan_setup_info)
+        setup_vlan(rand_selected_dut, vlan_setup_info, duthost, ptfhost)
         yield
     finally:
-        teardown_vlan(rand_selected_dut, vlan_setup_info)
+        _, _, _, portchannel_setup = vlan_setup_info
+        # Restore ptf configuration
+        ptfhost.set_dev_no_master(PTF_LAG_NAME)
+        for _, lag_data in portchannel_setup.items():
+            for idx in lag_data['ptf_member']:
+                ptf_lag_member = 'eth%u' % idx
+                ptfhost.set_dev_no_master(ptf_lag_member)
+                ptfhost.set_dev_up_or_down(ptf_lag_member, True)
+
+        ptfhost.shell("ip link del {}".format(PTF_LAG_NAME))
+        ptfhost.ptf_nn_agent()
```

Comment:
What's the purpose for calling `ptfhost.ptf_nn_agent` ?

Reply:
It's from test_lag_member:
https://github.com/sonic-net/sonic-mgmt/blob/master/tests/pc/test_lag_member.py#L107-L125